airflow/utils/logging.py

```diff
@@ -60,7 +60,8 @@ class S3Log(object):
     def __init__(self):
         remote_conn_id = configuration.get('core', 'REMOTE_LOG_CONN_ID')
         try:
-            from airflow.hooks.S3_hook import S3Hook
+            from airflow.contrib.hooks.s3_hook import S3Hook
```

Comment:
Mmm, it doesn't seem to be good practice to start relying on contrib in core.

@aoen should we upgrade the s3 hook in core?